mmv1/templates/terraform/custom_flatten/secret_version_version.go.erb

```diff
@@ -0,0 +1,25 @@
+<%# The license inside this block applies to this file.
+    # Copyright 2022 Google Inc.
+    # Licensed under the Apache License, Version 2.0 (the "License");
+    # you may not use this file except in compliance with the License.
+    # You may obtain a copy of the License at
+    #
+    #     http://www.apache.org/licenses/LICENSE-2.0
+    #
+    # Unless required by applicable law or agreed to in writing, software
+    # distributed under the License is distributed on an "AS IS" BASIS,
+    # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    # See the License for the specific language governing permissions and
+    # limitations under the License.
+-%>
+func flatten<%= prefix -%><%= titlelize_property(property) -%>(v interface{}, d *schema.ResourceData, config *Config) interface{} {
+    name := d.Get("name").(string)
+    secretRegex := regexp.MustCompile("projects/(.+)/secrets/(.+)/versions/(.+)$")
+
+    parts := secretRegex.FindStringSubmatch(name)
+    if len(parts) != 4 {
+        panic(fmt.Sprintf("Version name does not fit the format `projects/{{project}}/secrets/{{secret}}/versions/{{version}}`"))
```

Comment:
Hey @mitj04, thanks for the PR here. Can you help me understand the motivation behind this custom flatten?

I see that we have a panic added to the import logic, but it's another thing to panic here. The import logic is only run during `terraform import` which is much more likely run manually, and can be reacted to. A panic during the flatten will be run on every refresh, which could be bad in automated systems. Did you add this just for consistency with the import logic? If so, I think we should remove it for safety, even if the logic makes sense.

Reply:
Hi @c2thorn 

Yes, maintaining the consistency was one of the reason behind keeping it. But the primary motive was just to add a precautionary step. The if condition will (almost) never be executed unless there are changes in the API. As we know the `secret` attribute will always be in the format of projects/*/secrets/*/versions/*. We can blindly take the `versionId` knowing that it will always be present but just to be on the safe side I have added the if check.

We can replace it with `diags.Error` instead of panic. The reason we did not add it in the first place is that flatten will be called from the read call as well, and it is a best practice not to throw errors from the read call. If the `secret` is not in the proper format (...versions/*) then it would be inconsistent behaviour rather than an error so we thought it would be better to add panic for this case.

Kindly confirm once if you still believe we should remove the panic.

Thanks.

Reply:
Thanks for the clarification. I think I also missed the provider would crash anyway if  `parts[3]` wasn't available. Returning to this now, I think we are good to keep the panic.